stdlib/importlib/metadata/__init__.pyi

```diff
@@ -233,14 +241,14 @@ class MetadataPathFinder(DistributionFinder):
     @classmethod
     def find_distributions(cls, context: DistributionFinder.Context = ...) -> Iterable[PathDistribution]: ...
     if sys.version_info >= (3, 10):
-        # Yes, this is an instance method that has argumend named "cls"
+        # Yes, this is an instance method that has an argument named "cls"
         def invalidate_caches(cls) -> None: ...
 
 class PathDistribution(Distribution):
-    _path: Path
-    def __init__(self, path: Path) -> None: ...
-    def read_text(self, filename: StrPath) -> str: ...
-    def locate_file(self, path: StrPath) -> PathLike[str]: ...
+    _path: _SimplePath
+    def __init__(self, path: _SimplePath) -> None: ...
+    def read_text(self, filename: StrPath) -> str | None: ...
```

Comment:
Considering that neither `SimplePath.read_text()` nor `pathlib.Path.read_text()` can return `None`, I think just `str` is the better annotation here.

Reply:
It returns None if accessing the path throws an exception: https://github.com/python/cpython/blob/53d5e67804227d541ed2f9e8efea8de5d70cb1ec/Lib/importlib/metadata/__init__.py#L812

Reply:
You're right. I was looking at the 3.11 branch, where it doesn't have the explicit `return None`, but it still returns `None` implicitly.